### PR TITLE
Add Izhikevich Metabolic Nodes, clamp and fixed examples

### PR DESCRIPTION
**Pull request by Hananel-Hazan**
1. Add Izhikevich Metabolic Nodes and remove unnecessary parameters in Izhikevich and IzhikevichMetabolic
2. Change the behavioral of clamp, now clamp is forcing certain voltage value in specific neuron at certain time. 
3. Fix examples in Network to reflect the last changes in BindsNET improvements

### CODE REVIEW

File: bindsnet/network/__init__.py [hunk 1]
```diff
@@ -241,17 +245,19 @@ def run(self, inpts: Dict[str, torch.Tensor], time: int, **kwargs) -> None:
...
         # Simulate network activity for `time` timesteps.
         for t in range(timesteps):
-            # Update each layer of nodes.
+
+            # iterate on each layer
             for l in self.layers:
+                # Force neurons to certain voltage.
+                clamp = clamps.get(l, None)
+                if clamp is not None:
+                    self.layers[l].v = torch.where(clamp[t,:], clamps_v[l][t,:], self.layers[l].v)
```
Comment (djsaunde) [inline bindsnet/network/__init__.py]: Should be `self.layers[l].v = torch.where(clamp, clamps_v[l], self.layers[l].v`; the implementation of `clamps` (thus far) assumes there is no time dimension.

File: bindsnet/network/nodes.py [hunk 2]
```diff
@@ -703,3 +697,80 @@ def reset_(self) -> None:
...
+            self.c = -65.0 + 15 * (self.r ** 2)
+            self.d = 8 - 6 * (self.r ** 2)
+        else:
+            self.r = torch.rand(n)
+            self.a = 0.02 + 0.08 * self.r
+            self.b = 0.25 - 0.05 * self.r
+            self.c = -65.0 * torch.ones(n)
+            self.d = 2 * torch.ones(n)
+
+        m = 0.7
```
Comment (djsaunde) [inline bindsnet/network/nodes.py]: Should be `self.m = 0.7`; otherwise, it can't be accessed elsewhere.

File: bindsnet/network/__init__.py (hunk 1 above)
Comment (Hananel-Hazan) [reply bindsnet/network/__init__.py]: That right, now clamp have time dimension

File: bindsnet/network/nodes.py (hunk 2 above)
Comment (Hananel-Hazan) [reply bindsnet/network/nodes.py]: You right

File: bindsnet/network/__init__.py (hunk 1 above)
Comment (djsaunde) [reply bindsnet/network/__init__.py]: Why? The input is passed in, timestep by timestep.

Also, this breaks some of the examples.

**Pull request merged**